src/components/forms/JurisdictionMetadata/index.tsx

```diff
@@ -0,0 +1,230 @@
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import Papaparse from 'papaparse';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import { toast } from 'react-toastify';
+import { Button, Label } from 'reactstrap';
+import { FormGroup } from 'reactstrap';
+import * as Yup from 'yup';
+import {
+  CSV_FILE,
+  FILE,
+  FILE_UPLOADED_SUCCESSFULLY,
+  JURISDICTION_ID,
+  JURISDICTION_METADATA,
+  JURISDICTION_NAME,
+  REQUIRED,
+  UPLOAD,
+  UPLOAD_FILE,
+  UPLOADING,
+} from '../../../configs/lang';
+import { HOME_URL, OPENSRP_V1_SETTINGS_ENDPOINT } from '../../../constants';
+import { growl } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+const SUPPORTED_FORMATS = ['text/csv'];
+
+/** yup validation schema for Jurisdiction Metadata Form input */
+export const JurisdictionSchema = Yup.object().shape({
+  file: Yup.mixed()
+    .required(REQUIRED)
+    .test('fileFormat', 'CSV Files only', value => value && SUPPORTED_FORMATS.includes(value.type)),
+});
+
+export interface JurisdictionMetadataFormFields {
+  file: File;
+}
+
+export interface Setting {
+  description: string;
+  label: string;
+  value: string | unknown;
+  key: string;
+}
+
+export interface SettingConfiguration {
+  type: string;
+  identifier: string;
+  providerId: string;
+  locationId: string;
+  settings: Setting[];
+  teamId: string;
+}
+
+export interface JurisdictionMetadataFormProps {
+  disabledFields: string[];
+  OpenSRPService: new (...args: any[]) => any;
+  submitForm: (
+    setSubmitting: (isSubmitting: boolean) => void,
+    setGlobalError: (errorMessage: string) => void,
+    setIfDoneHere: (closeSubmissionCycle: boolean) => void,
+    props: JurisdictionMetadataFormProps,
+    values?: JurisdictionMetadataFormFields
+  ) => void;
+  initialValues: JurisdictionMetadataFormFields;
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: JurisdictionMetadataFormFields = {
+  file: new File([], ''),
+};
+
+// create payloads for sending
+const createPayloads = (result: any): SettingConfiguration[] => {
+  const payloads: SettingConfiguration[] = [];
+  const { data } = result;
+  // check if jurisdiction_id exists
+  if (data.length > 0 && data[0].jurisdiction_id) {
+    // get the metadata items
+    const headers = Object.keys(data[0]);
+    for (const header of headers) {
+      const settings: Setting[] = [];
+      if (header !== JURISDICTION_ID && header !== JURISDICTION_NAME) {
```

Comment:
Maybe we can instead use an array of headers to ignore that would then be a param that this function takes, with the default being [JURISDICTION_ID, JURISDICTION_NAME]